stl/inc/valarray

```diff
@@ -555,10 +555,12 @@ private:
 
     void _Tidy_deallocate() noexcept {
         if (_Myptr) { // destroy elements
-            for (size_t _Idx = 0; _Idx < _Mysize; ++_Idx) {
-                _Destroy_in_place(_Myptr[_Idx]);
+            if constexpr (!is_trivially_destructible_v<_Ty>) {
+                const size_t _Size = _Mysize; // eliminating indirection helps vectorization
+                for (size_t _Idx = 0; _Idx < _Size; ++_Idx) {
+                    _Destroy_in_place(_Myptr[_Idx]);
+                }
             }

```

Comment:
Could we simply:
```suggestion
            _Destroy_range(_Myptr, _Myptr + _Mysize);
```
here? It already optimizes for trivially destructible types.

Reply:
I believe this is a strict improvement, yes. Good catch - `valarray` predates `_Destroy_range` and we never noticed we could call it here.